### PR TITLE
add one-click harvesting configuration #1820

### PR DESCRIPTION
**Pull request by ishidad2**
### What the current behaviour is

- Currently, harvesting in mobile and desktop wallets is working in two steps.  #1820 

### Why this is an issue

- These two transactions can be combined into one aggregate transaction

### What's the fix

1. Create Account Links (Remote, VRF, Node).
2. Send a persistent delegated message to Node.

These two transactions are combined into one and sent as an aggregate transaction.

### CODE REVIEW

**Comment by ishidad2** [comment]
I was able to reproduce it.

I had forgotten that there was a key-link-only condition in the old wallet, since the new version of the wallet does not generate a key-link-only condition.

I first reverted the "Key Link" tab back to its original state to address this issue.

Also, as another way to approach this I would consider the following
Please give me your opinion.

Issue an aggregate transaction with the unlinked transaction also included in the harvest setting.

I believe that doing the above would require a slight change in operations.
(Like Arcana wallet, when selecting a node, make an aggregate transaction that includes "LinkAction.Unlink", "LinkAction.Link", and "PersistentDelegationRequestTransaction").

In other words, the "Stop Harvesting" operation should no longer be necessary.

I believe it will take some time to achieve this, as it will require some changes in the UI.

What do you think?

**Comment by cryptoBeliever** [comment]
@ishidad2 
Current logic is very complicated (FormPersistentDelegationRequestTransactionTs.ts) and needs to be analyzed in code (I think many conditions can be removed when we join those two steps) but I think we need :

1) Start harvesting:

Start harvesting should send one aggregate transaction containing:

- unlink remote account (only if an account is already linked and we don't have a private key for it in storage)
- unlink VRF key (only if an account is already linked and we don't have a private key for it in storage)
- unlink node key (if a node key is linked - we need to unlink the old one)
- link remote account (only if we had to remove the old one - then we have to generate and link a new key)
- link VRF key (only if we had to remove the old one - then we have to generate and link the new key)
- link node key (always)
- persistent delegated message (always)

2) Stop harvesting: 

Start harvesting should send one aggregate transaction containing:
- unlink node key (only if a node key is linked do we need to unlink the old one)
- unlink remote account (only if an account is already linked and we don't have the private key for it in storage)
- unlink VRF key (only if an account is already linked and we don't have the private key for it in storage)

@yilmazbahadir what do you think about it?

@ishidad2 when we decide on details you can update the issue description to provide more details of what changes introduce in this PR (here examples https://github.com/symbol/desktop-wallet/pull/1847, https://github.com/symbol/product/pull/117 - you can follow template: "What the current behaviour is", "Why this is an issue.", "What's the fix") .

**Comment by ishidad2** [comment]
### What the current behaviour is

- Currently, harvesting in mobile and desktop wallets is working in two steps.  #1820 


### Why this is an issue

- These two transactions can be combined into one aggregate transaction

### What's the fix

1. Create Account Links (Remote, VRF, Node).
2. Send a persistent delegated message to Node.

These two transactions are combined into one and sent as an aggregate transaction.

**Comment by yilmazbahadir** [comment]
Implemented with https://github.com/symbol/desktop-wallet/pull/1903
